src/write_gcm.c

```diff
@@ -158,7 +158,69 @@ static void EVP_MD_CTX_free(EVP_MD_CTX *ctx) {
   EVP_MD_CTX_cleanup(ctx);
   OPENSSL_free(ctx);
 }
-#endif
+
+// Multi-threading support for curl and TLS. This is needed because wg_write
+// creates one thread for each queue (ATS and GSD).
+// See https://curl.haxx.se/libcurl/c/threadsafe.html
+#include <openssl/crypto.h>
+
+static pthread_mutex_t *lockarray;
```

Comment:
[Optional] This could also be `crypto_locks`, right? Though that's a bit less of an issue, given that it's only used in this section.